pkg/odo/cli/component/create.go

```diff
@@ -946,7 +961,21 @@ func (co *CreateOptions) Run() (err error) {
 
 			if !util.CheckPathExists(DevfilePath) {
 				// Download devfile from registry
-				err := util.DownloadFile(co.devfileMetadata.devfileRegistry.URL+co.devfileMetadata.devfileLink, DevfilePath)
+				token := ""
```

Comment:
Nit: Same as my other nit, it might be cleaner to declare the params struct above the IsSecure check (without the token field) and then assign the value of `token` inside the if-statement. So something like:
```
token, err = keyring.Get(util.CredentialPrefix+registry.Name, "default")
if err != nil {
    return nil, errors.Wrap(err, "Unable to get secure registry credential from keyring")
}
request.Token = token
```

Reply:
Good suggestion, I will refactor it.